platform/pulpcore/app/views/content.py

```diff
@@ -0,0 +1,152 @@
+import os
+
+from django.core.exceptions import ObjectDoesNotExist
+from django.http import HttpResponseNotFound, StreamingHttpResponse
+from django.views.generic import View
+
+from wsgiref.util import FileWrapper
+
+from pulpcore.app.models import Distribution
+
+
+class ContentView(View):
+    """
+    Content endpoint.
+
+    URL matching algorithm.
+
+    http://redhat.com/cdn/stage/files/manifest
+                     |--------------||--------|
+                            (1)          (2)
+
+    1. Match: Distribution.base_path
+    2. Match: PublishedFile.relative_path
+    """
+
+    BASE_URL = '/content'
+
+    @staticmethod
+    def _base_paths(path):
+        """
+        Get a list of base paths used to match a distribution.
+
+        Args:
+            path (str): The path component of the URL.
+
+        Returns:
+            list: Of base paths.
+
+        """
+        tree = []
+        while True:
+            base, _ = os.path.split(path)
+            if not base:
+                break
+            tree.append(base.lstrip('/'))
+            path = base
+        return tree
+
+    def _match(self, path):
+        """
+        Match either a PublishedArtifact or PublishedMetadata.
+
+        Args:
+            path (str): The path component of the URL.
+
+        Returns:
+            str: The storage path of the matched object.
+
+        Raises:
+            ObjectDoesNotExist: The referenced object does not exist.
+            None: When not matched.
+
+        """
+        base_paths = self._base_paths(path)
+        distribution = Distribution.objects.get(base_path__in=base_paths)
+        publication = distribution.publication
+        rel_path = path.lstrip(distribution.base_path)
```

Comment:
What is your value of `base_path`?

The `base_path` is a _relative_ path and must not contain a leading '/'.  This is noted in the docstring for the Distribution model and hopefully would be enfored by the viewset.

@dkliban: FYI, I pushed an update that improves handling of 404 conditions.